### PR TITLE
Bug/doxygen cleanup

### PR DESCRIPTION
**Pull request by robszewczyk**

### CODE REVIEW

File: src/lib/profiles/data-management/Current/SubscriptionClient.h [hunk 1]
```diff
@@ -74,55 +74,102 @@ class SubscriptionClient
...
+         * subscribe to.
+         */
         kEvent_OnSubscribeRequestPrepareNeeded = 3,
 
-        // cancel, abort, and free are allowed
+        /**
+         * Generated upon a successful establishment of the subscription.  The
+         * application may call EndSubscription(), AbortSubscription(), or
+         * Free() at this point.
+        */
```
Comment (gerickson) [inline src/lib/profiles/data-management/Current/SubscriptionClient.h]: Minor nit: '*' alignment.

File: src/lib/profiles/data-management/Current/SubscriptionClient.h [hunk 2]
```diff
@@ -74,55 +74,102 @@ class SubscriptionClient
...
+         */
         kEvent_OnEventStreamReceived = 7,
 
-        // An event indicating subscription activity
+        /**
+         * An event indicating subscription activity.  It is generated whenever
+         * the client receives an indication on subscription activity; this
+         * includes successful subscription establishment, a reception of a
+         * NotificationRequest or a receipt of a status report in response to
+         * the SubscribeConfirm message. */
```
Comment (gerickson) [inline src/lib/profiles/data-management/Current/SubscriptionClient.h]: Minor nit: '*/' alignment.

File: src/lib/profiles/data-management/Current/SubscriptionClient.h (hunk 1 above)
Comment (robszewczyk) [reply src/lib/profiles/data-management/Current/SubscriptionClient.h]: done

File: src/lib/profiles/data-management/Current/SubscriptionClient.h (hunk 2 above)
Comment (robszewczyk) [reply src/lib/profiles/data-management/Current/SubscriptionClient.h]: done

**Pull request merged**